testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToHudi.java

```diff
@@ -0,0 +1,448 @@
+
+        createHudiCowTable(hudiTable.schemaTableName(), false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("GRANT SELECT ON " + hiveTable.fullName() + " TO ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTable.fullName() + " is redirected to " + hudiTable.fullName() + " and GRANT is not supported with table redirections");
+
+        dropHudiTable(hudiTable);
+    }
+
+    @Test(groups = {HIVE_HUDI_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testRevoke()
+    {
+        String tableName = "hudi_revoke_" + randomNameSuffix();
+        TestTableName hudiTable = new TestTableName("hudi", "default", tableName);
+        TestTableName hiveTable = new TestTableName("hive", "default", tableName);
+
+        createHudiCowTable(hudiTable.schemaTableName(), false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("REVOKE SELECT ON " + hiveTable.fullName() + " FROM ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTable.fullName() + " is redirected to " + hudiTable.fullName() + " and REVOKE is not supported with table redirections");
+
+        dropHudiTable(hudiTable);
+    }
+
+    @Test(groups = {HIVE_HUDI_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testSetTableAuthorization()
+    {
+        String tableName = "hudi_set_table_authorization_" + randomNameSuffix();
+        TestTableName hudiTable = new TestTableName("hudi", "default", tableName);
+        TestTableName hiveTable = new TestTableName("hive", "default", tableName);
+
+        createHudiCowTable(hudiTable.schemaTableName(), false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("ALTER TABLE " + hiveTable.fullName() + " SET AUTHORIZATION ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTable.fullName() + " is redirected to " + hudiTable.fullName() + " and SET TABLE AUTHORIZATION is not supported with table redirections");
+
+        dropHudiTable(hudiTable);
+    }
+
+    @Test(groups = {HIVE_HUDI_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testDeny()
+    {
+        String tableName = "hudi_deny_" + randomNameSuffix();
+        TestTableName hudiTable = new TestTableName("hudi", "default", tableName);
+        TestTableName hiveTable = new TestTableName("hive", "default", tableName);
+
+        createHudiCowTable(hudiTable.schemaTableName(), false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("DENY DELETE ON " + hiveTable.fullName() + " TO ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTable.fullName() + " is redirected to " + hudiTable.fullName() + " and DENY is not supported with table redirections");
+
+        dropHudiTable(hudiTable);
+    }
+
+    private static void dropHudiTable(TestTableName tableName)
```

Comment:
I would inline this method.

Reply:
could you please elaborate a reason of inlining. the reason of extract is to make code more readable by removal repeatable SQL constructions.

Reply:
inlined